### PR TITLE
feat: relation between albums & medias

### LINKED ISSUE #5: See which photos belong to what album

### PR DESCRIPTION
**Pull request by siriusnottin**
This PR modifies code on several files to replace existing Google Photos API requests with calls to methods of `GPhotos` class, improving consistency and organization.

**You can now link albums and medias.** (closes #5 )

- Changes the `search` method in `mediaItems` class to support `fields` parameter (better performance when fetching only `mediaItems ids`).
- Also tweaks the `getMediaItemsFromAlbum` function to use the new `mediaItems.search` method.
- Finally, refactored the `Album` interface using a new `MediaItemReference` interface.

### Todo

Needs more stuff tbd before merging:

- [x] Refactor `SyncAlbums`
- [x] Link Albums and Medias
- [x] `favorite` filter on Medias SyncTable
- [x] TypeScript warning on execute method
- [x] Better error handling
- [x] See comments

### CODE REVIEW

**Comment by siriusnottin** [comment]
### Todo

Needs more stuff tbd before merging:

- [ ] TypeScript warning on execute method
- [ ] `favorite` filter on Medias SyncTable
- [ ] Check for unused imports
- [ ] Check types

File: pack.ts [hunk 1]
```diff
@@ -69,13 +70,26 @@ pack.addSyncTable({
...
+      // Feature filter (favorites)
+      if (favorite) {
+        const featureFilter = new photos.FeatureFilter(types.MediaFeature.Favorites)
+        filters.setFeatureFilter(featureFilter);
+      }
+
+      const response = await photos.mediaItems.search(
+        filters,
+        '',
+        100,
```
Comment (siriusnottin) [inline pack.ts:85]: I don't like the way the params are passed here. Maybe put them in an object with props so it's more clear?

File: pack.ts [hunk 2]
```diff
@@ -69,13 +70,26 @@ pack.addSyncTable({
...
+        (context.sync.continuation?.nextPageToken as string | undefined)
+      )
```
Comment (siriusnottin) [inline pack.ts:87]: I don't like the way the params are passed here. Maybe put them in an object with props so it's more clear?

File: pack.ts [hunk 3]
```diff
@@ -69,13 +70,26 @@ pack.addSyncTable({
...
+      const { nextPageToken, mediaItems } = response?.body ?? {};
```
Comment (siriusnottin) [inline pack.ts:88]: is response really optional here?

File: pack.ts [hunk 4]
```diff
@@ -90,7 +104,24 @@ pack.addSyncTable({
     description: "Sync all albums.",
     parameters: [],
     execute: async function ([], context) {
-      return helpers.syncAlbums(context);
+
+      const photos = new GPhotos(context);
+      const response = await photos.albums.list(20, (context.sync.continuation?.nextPageToken as string | undefined));
+      const { albums, nextPageToken } = response?.body;
```
Comment (siriusnottin) [inline pack.ts:110]: same, is response optional?

File: pack.ts (hunk 3 above)
Comment (siriusnottin) [reply pack.ts:88]: Need to test with an empty GP library

File: pack.ts (hunk 2 above)
Comment (siriusnottin) [reply pack.ts:87]: Will be done in the next PR

File: pack.ts (hunk 3 above)
Comment (siriusnottin) [reply pack.ts:88]: Will be done when I'll add some tests
